lively.project/project.js

```diff
@@ -33,10 +35,44 @@ export class Project {
     return localInterface.coreInterface;
   }
 
+  static async ensureGitResource (fullName) {
+    return resource('git/' + await defaultDirectory()).join('..').join('local_projects').join(fullName).withRelativePartsResolved().asDirectory();
+  }
+
+  static fetchInfoPreflight (fullName) {
+    let finishCheck;
+    ({ promise: lively.checkedProjectVersionRelation, resolve: finishCheck } = promise.deferred());
```

Comment:
Just for my understanding - what happens here is that we store a promise inside of `lively.checkedProjectVersionRelation` so that we can do `await lively.checkedProjectVersionRelation` until the resolve callback is called, right?

The resolve callback in this case is called `finishCheck` and, despite that not really being transparent anywhere - that is a function and the `then` chain will also take care of calling it for us (despite the last entry being `finishCheck` and not `finishCheck()`) which will let anything waiting for `lively.checkedProjectVersionRelation` continue?